### PR TITLE
Update to OTel 0.14.1

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpGrpcSpanExporterProperties.java [hunk 1]
```diff
@@ -5,8 +5,8 @@
 
 package io.opentelemetry.instrumentation.spring.autoconfigure.exporters.otlp;
 
-import static io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter.DEFAULT_DEADLINE_MS;
 import static io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter.DEFAULT_ENDPOINT;
+import static io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter.DEFAULT_TIMEOUT_MS;
```
Comment (anuraaga) [inline instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpGrpcSpanExporterProperties.java:9]: @jkwatson Looks like I accidentally breaking-changed this field during some WIP on the timeout PRs

File: instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java [hunk 2]
```diff
@@ -7,60 +7,77 @@
...
-      String metricName, String unit, MetricData.Type type, boolean checkNonZeroValue) {
+      String metricName, String unit, MetricDataType type, boolean checkNonZeroValue) {
     List<MetricData> metricDataList = testMetricExporter.metricDataList;
     for (MetricData metricData : metricDataList) {
       if (metricData.getName().equals(metricName)) {
         assertThat(metricData.getDescription()).isNotEmpty();
         assertThat(metricData.getUnit()).isEqualTo(unit);
-        assertThat(metricData.getPoints()).isNotEmpty();
+        metricData.getDoubleGaugeData().getPoints();
+        List<Point> points = new ArrayList<>();
```
Comment (anuraaga) [inline instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java:62]: @jkwatson Don't know if it matters, but the impact of losing `getPoints` was relatively large. Presumably I should not be instanceof below, but without doing a deeper dive into this test it's not easy to know what's a long gauge or a long sum so used this pattern.

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by pavolloffay** [comment]
@anuraaga @trask @iNikem roughly when the 0.14.0 will be released?

File: instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java (hunk 2 above)
Comment (jkwatson) [reply instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java:62]: paging @bogdandrutu

File: instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpGrpcSpanExporterProperties.java (hunk 1 above)
Comment (jkwatson) [reply instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpGrpcSpanExporterProperties.java:9]: do you think we need a patch to address it?

File: instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java (hunk 2 above)
Comment (jkwatson) [reply instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java:62]: This doesn't look like a huge impact to me. Am I missing something?

File: instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java [hunk 3]
```diff
@@ -7,60 +7,77 @@
...
   public void verify(
-      String metricName, String unit, MetricData.Type type, boolean checkNonZeroValue) {
+      String metricName, String unit, MetricDataType type, boolean checkNonZeroValue) {
     List<MetricData> metricDataList = testMetricExporter.metricDataList;
     for (MetricData metricData : metricDataList) {
       if (metricData.getName().equals(metricName)) {
         assertThat(metricData.getDescription()).isNotEmpty();
         assertThat(metricData.getUnit()).isEqualTo(unit);
-        assertThat(metricData.getPoints()).isNotEmpty();
+        metricData.getDoubleGaugeData().getPoints();
```
Comment (bogdandrutu) [inline instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java:61]: This is a no-op line?

File: instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java (hunk 2 above)
Comment (bogdandrutu) [reply instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java:62]: I would actually merge this logic with the next 20 lines:

switch metric.getType() {
  case GaugeDouble:
    List<DoublePoints> dpts = metricData.getDoubleGaugeData().getPoints();
    dpts.foreach(doublePoint -> assertThat(doublePoint.getValue()).isGreaterThan(0.0));
  ....
}

Or something like that (written from the phone so no access to IDE 👍 )

**Review by trask — APPROVED** [review]
thanks!! 😅

File: instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpGrpcSpanExporterProperties.java (hunk 1 above)
Comment (anuraaga) [reply instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpGrpcSpanExporterProperties.java:9]: Nah don't think so we should just update the release notes

File: instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java (hunk 2 above)
Comment (anuraaga) [reply instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java:62]: I would have to understand the test much better to e.g. understand whether to use gauge vs sum. Will go with my brain-dead change for now and hopefully someone can give some maintenance to the test, maybe after the metrics stuff stabilizes some more.

**Pull request merged**